tests/gilmour_proxy_test.go

```diff
@@ -0,0 +1,304 @@
+package tests
```

Comment:
If you are testing the proxy package, the tests should also be in the proxy package. Also, the test file should be called `<filename>_test.go`